### DISCUSSION TITLE
Metrics SVG on Portfolio Website

### DISCUSSION THREAD

**Discussion by DalpatRathore** [lowlighter/metrics#871] [Q&A] [answered]
### Metrics SVG on Portfolio Website
  &nbsp; &nbsp; &nbsp; I want to use cache `metrics` SVG on portfolio webpage, which automatically get updated like GitHub Actions and cache SVG for an hour or two.
 - **Shared Instance** only cache an image for an hour and is also not always available.
 - **Github Actions** - don't provide URL of  `metrics` SVG. Though I deployed this on one of the repo and working fine.

**Comment by lowlighter** [comment, marked as answer]
Hi 👋 !

Using GitHub Actions should work, just use the "raw" url of the generated image on your website, like:

```html
<img src="https://raw.githubusercontent.com/lowlighter/metrics/examples/metrics.plugin.tweets.attachments.svg">
```

It should work, I remember doing it in the past

Hope it helps 👍

**Reply by DalpatRathore** [reply]
Hi! @lowlighter, thanks for the suggestion. Though I have tried this same process before posting the query. It gives SVG without icon image and also I have to make that repo public.

- SVG 
 ![Capture](https://user-images.githubusercontent.com/69510006/153698927-dc779879-7948-43b6-9029-ee8ac179adff.JPG)
-  Q. If I use that raw image link on the webpage will that SVG update automatically?

**Reply by lowlighter** [reply]
> It gives SVG without icon image

Even if images does not appear in the raw link I think they should be visible when you include them on your website. It's just a display issue from viewing raw svgs in the browser (I don't remember the exact reason but I think it was something linked to default cross-origin policy)

If it doesn't, you could try to turn off [`config_base64`](https://github.com/lowlighter/metrics/tree/master/source/plugins/core#config_base64) so images aren't encoded as base64.

> I have to make that repo public

If you really need to hide the repository, you could use [`config_output: gist`](https://github.com/lowlighter/metrics/tree/master/source/plugins/core#using-gists) along with `committer_gist` to use GitHub gists instead, and use uploaded files instead (that's actually what I've done on my personal repository)

You could also just use [`output_action: none`](https://github.com/lowlighter/metrics/blob/master/source/plugins/core/README.md#manual-handling) to just manually handle the results and do what you want with it if nothing here satisfy your use case

> Q. If I use that raw image link on the webpage will that SVG update automatically?

Yes, they'll be updated but be sure to remove the permalink (i.e. sha/commit) from the url else it'll stay stuck at a given revision x)

**Reply by DalpatRathore** [reply]
:zap:  Thanks for resolving my query and explaining it with clarity, orderly, and organized manner. :pray: 

:zap: Thanks also for building this amazing and very useful ***Github Profile Metrics***. :heart: :heart: :heart:

**Reply by lowlighter** [reply]
Thanks for your support!
